\documentclass[12pt, oneside]{article}   	% use "a msart" instead of "article" for AMSLaTeX format
\usepackage{color}
\usepackage{geometry}                		% See geometry.pdf to learn the layout options. There are lots.
\geometry{letterpaper}                   		% ... or a4paper or a5paper or ... 
%\geometry{landscape}                		% Activate for for rotated page geometry
%\usepackage[parfill]{parskip}    		% Activate to begin paragraphs with an empty line rather than an indent
\usepackage{graphicx}				% Use pdf, png, jpg, or epsÂ§ with pdflatex; use eps in DVI mode
								% TeX will automatically convert eps --> pdf in pdflatex		
\usepackage{amssymb}
\usepackage{amsmath}
\usepackage[compact]{titlesec}
\linespread{1.7}
\usepackage{float}
\usepackage{pdflscape}
%\usepackage{rotating}
\usepackage{soul}
\usepackage{longtable}
\usepackage{caption,setspace}
\captionsetup{font={stretch=1.0}}
%\usepackage{threeparttable}
\usepackage{lineno}
\usepackage[round]{natbib} %round makes parentheses instead of square brackets
\usepackage{url}
%\usepackage{authblk}
\setcounter{secnumdepth}{4}
\titleformat{\paragraph}
{\normalfont\normalsize\bfseries}{\theparagraph}{1em}{}
\titlespacing*{\paragraph}
{0pt}{3.25ex plus 1ex minus .2ex}{1.5ex plus .2ex}
\graphicspath{ {images/} }

\author{}
\author{Allison G.\ Dedrick$^{a, \ast}$ \\
Katrina A.\ Catalano$^a$ \\
Michelle R.\ Stuart$^a$ \\
J.\ Wilson White$^b$ \\
Humberto Montes, Jr.\ $^c$ \\
Malin Pinsky$^a$}

\title{Clownfish metapopulation persistence draft}

\date{} 

\begin{document}
\renewcommand{\topfraction}{0.95}
\maketitle{}

\noindent{} a. Department of Ecology Evolution and Natural Resources, Rutgers University, 14 College Farm Road, New Brunswick, NJ 08901;

\noindent{} b. Oregon State University

\noindent{} c. Visayas State University

\noindent{} $\ast$ Corresponding author; e-mail: agdedrick@gmail.com

(\textit{Author order not yet determined})

%\noindent{} 1. Present address: Rutgers University, 14 College Farm Rd., New Brunswick, NJ, USA 08901.

\bigskip

%\section*{Abstract}

% If submit to Ecology Letters, max 5000 words, 6 figures/tables/text boxes

\linenumbers{}
\modulolinenumbers[3]

\section*{Introduction}

Metapopulations exist along a continuum, with dynamics driven by the balance of extinction and colonization of local patches at one extreme and focused on the balance of immigration and emmigration at constantly-occupied local patches at the other \citep{kritzer2006marine}. Terrestrial metapopulations often show extinction-colonization dynamics \citep[e.g.][]{hanski1998metapopulation}, while marine metapopulations tend to exhibit immigration-emmigration dynamics where local extinction of patches is uncommon \citep{kritzer2006marine}. For these marine metapopulations, dynamics and persistence depend on connectivity among patches and the demographic rates at each patch \citep[e.g.][]{hastings_persistence_2006, hanski1998metapopulation}. Assessing levels of connectivity and demographic parameters has been particularly challenging for marine species, where much of the mortality and movement happens at larval and juvenile stages when individuals are hard to track and have the potential to travel long distances with ocean currents \citep[e.g.][]{kritzer2006marine, cowen_larval_2009, roughgarden_recruitment_1988}. A need to understand metapopulations for conservation and management, such as siting marine protected areas \citep[e.g.][]{botsford_dependence_2001,white_population_2010}, however, has led to a large body of theory describing how marine metapopulations might persist. 

For any population to persist, individuals must on average replace themselves during their lifetime. Assessing replacement must take into account the demographic processes across the whole life cycle, including how likely individuals are to survive to the next age or stage, their expected fecundity at each stage, and the survival of any offspring produced to recruitment. In a spatially-structured population, as many marine populations are, in addition to assessing whether the reproductive output and survival of a population is sufficient, we must also consider how the offspring are distributed across space. Marine larvae were once thought to be well-mixed and dispersed far on ocean currents \citep[e.g.][]{roughgarden_recruitment_1988}, suggesting widespread connectivity among patches and largely open populations. Recent advances in estimating connectivity through natural tags and genetics, however, suggest that dispersal may be more limited \citep[e.g.][]{daloia_self-recruitment_2013, hameed2016inverse, almany2017larval}, and local persistence of marine populations on a small spatial scale is seeming more possible.  % Don't love this end sentence... Does this paragraph tie together cohesively enough?

Considering both the demographic processes within patches and the connectivity among them, a metapopulation can persist in two ways: 1) at least one patch can achieve replacement in isolation, or 2) patches receive enough recruitment to achieve replacement through multi-generational loops of connectivity with other patches in the metapopulation \citep{hastings_persistence_2006, burgess2014beyond}. In the first case (termed self-persistence), enough of the reproductive output produced at one patch is retained at the patch for it to persist. In the second (network persistence), closed loops of connectivity among at least some of the patches - where individuals from one patch settle at another and eventually send offspring back to the first in a future generation - provide the patch with enough recruitment to persist within the network. Though it has been challenging to estimate the parameters necessary to understand how actual metapopulations persist, a large work of theory developed in part to guide marine protected area design helps predict when each type of persistence is likely to occur \citep[i.e., large patches relative to the mean dispersal distance are likely to be self-persistent,][]{botsford_dependence_2001}. % Is this sentence relevant here? %If one patch persists, the metapopulation persists.

New ways of identifying individuals and determining their origins, such as otolith and shell microchemistry and genetic parentage analysis \citep[e.g.][]{wang2004sibship, wang2014estimation} are making it increasingly possible to estimate both the demographic \citep[e.g.][]{carson2011evaluating, hameed2016inverse} and the dispersal \citep[e.g.][]{almany2017larval, daloia_self-recruitment_2013} parameters necessary to assess persistence in real metapopulations. We might expect that populations on isolated islands are the most likely to be self-persistent, as they lack nearby populations with which to exchange larvae and would go locally extinct if they did not achive replacement. At isolated Kimbe Island in Papua New Guinea, \cite{salles_coral_2015} find that the population of orange clownfish (\textit{Amphiprion percula}) can likely persist without outside immigration. In contrast, populations of bicolor damselfish (\textit{Stegastes partitus}) at a set of reef patches across four isolated islands in the Bahamas do not appear able to persist without outside input \citep{johnson2018integrating}. For populations that exist in patches along a continuous linear coastline, rather than on separate islands, however, how patches interact and what the scale of metapopulation persistence is are still open questions. %Flow differences along extended coastlines compared to isolated islands likely change the patterns of connectivity and ability of larvae to stay close to their natal site (NEED CITATIONS, EXPLANATION OF FLOW), possibly changing the scale of a metapopulation. % Might be more likely to have flow dominated by currents on coastlines and eddies around remote islands? Is this a discussion point instead?

The number of studies estimating demographic rates and connectivity in marine metapopulations is growing \citep[e.g.][]{salles_coral_2015, johnson2018integrating, garavelli2018population}, but most use data from one or a few years. Longer data sets enable better estimates of long-term average rates, rather than assuming the demographic and dispersal rates from a particular year or two are representative through time. More data is also useful for explicitly considering uncertainty, both to assess how well we understand persistence for a population and to see which parameters contribute most to our uncertainty. Finally, sampling over many years provides the possibility of comparing abundance trends to persistence metrics to see if they tell a consistent story.

% Characterizing and understanding variability is important as it can drive dynamics, even allowing persistence when average rates would not (SEBASTIAN CITATIONS), but using data across several years to get a sense of the range of rates and an average is a first step. Having multiple years of data can provide a range of estimates, rather than one point value, and help us understand which parameters contribute most to uncertainty in persistence. 

Here, we further our understanding of metapopulation dynamics in a network of patches along a coastline through a study of yellowtail clownfish (\textit{Amphiprion clarkii}) in the Philippines. We assess persistence for all patches of habitat within a 30 km stretch of coastline, which exceeds estimates of the dispersal spread for this species \citep{pinsky2010using}, suggesting the network is likely to operate as a contained metapopulation. With seven years of annual sampling data, we are able to estimate persistence metrics and replacement over the longer term and investigate abundance through time to compare with the replacement-based persistence metrics. We use our long-term data set from habitat patches on a continuous section of coastline to understand persistence within a local network. % Work on last sentence...

\section*{Methods} 

\subsection*{Persistence theory and metrics}

For a population to persist, individuals must be able to replace themselves on average at low abundance \citep[e.g.][]{hastings_persistence_2006,botsford_connectivity_2009}. In non-spatially structured populations, we use criteria such as the average number of recruiting offspring each individual produces during its life (called $R_0$ when the population is age-structured and density-independent) or the growth rate of the population (such as the dominant eigenvalue $\lambda$ of an age-structured Leslie matrix) \citep{caswell_matrix_2001, burgess2014beyond}. For spatially-structured populations, we must also consider the spatial spread of offspring, often represented through a dispersal kernel or connectivity matrix \citep[e.g.][]{cowen_scaling_2006, buston2011probability, hogan_local_2011, daloia2015patterns}). 

We consider three primary metrics to assess whether and how the population is persistent: 1) lifetime recruit production (LRP), to assess whether the population has enough surviving offspring to achieve replacement 2) self-persistence (SP), to assess whether any individual patch can persist in isolation without input from other patches, and 3) network persistence (NP), to assess whether the metapopulation is persistent as a connected unit. We explain each metric below in detail. To represent the uncertainty in our estimates, we calculate each metric 1000 times, pulling each input parameter from a distribution or range. In our results, we show the range of values of each persistence metric as well as our best estimate. % Should this have the \% Reference the section with \ref?

\paragraph*{Lifetime production of recruits}

We find the estimated number of recruits an individual recruit will produce (lifetime recruit production: LRP) by multiplying the total number of eggs a recruit-sized individual will produce in its lifetime (lifetime egg production: LEP) by the fraction of those eggs that will survive to become recruits (egg-recruit survival: $S_e$) (Fig.\ \ref{FIG_Schematic} Metrics):
\begin{equation}
\text{LRP} = \text{LEP} * S_e. \label{EQN_LRP}
\end{equation}
If $LRP \geq 1$, the population has the possibility for replacement; indviduals produce enough surviving offspring, before taking into account the probability of dispersal. If $LRP < 1$, the individuals are not replacing themselves and the population cannot persist without input from outside patches. We consider LRP for all recruits produced by our individuals, regardless of where they settle, which requires combination with the dispersal kernel to understand persistence, and for recruits that settle only at our sites, $\text{LRP}_\text{local}$, which implicitly includes dispersal. If $\text{LRP}_\text{local} \geq 1$, our group of sites is able to persist locally. 

\paragraph*{Self-persistence} 

A patch is able to persist in isolation (self-persistent) if individuals produce enough offspring that survive to recruitment (LRP) and disperse back to the natal patch (with probability of dispersal $p_{i,i}$) to replace themselves. \cite{burgess2014beyond} use LEP to represent offspring produced and local retention (LR) - the number of surviving recruits that disperse back to the natal patch over the number of eggs produced by the natal patch - to capture egg-recruit survival and dispersal combined in a criteria for self-persistence: $\text{LEP} \times \text{LR} \geq 1$. We modify this to include egg-recruit survival in the offspring term, using LRP in place of LEP, to assess whether a particular patch $i$ is self-persistent:

\begin{equation}
\begin{split}
%SP_i &= \text{LEP} \times \text{LR} \\
SP_i &= \text{LEP} \times \frac{\text{recruits}}{\text{egg}} \times \frac{p_{i,i} \times \text{\# recruits from patch i}}{\frac{\text{recruits}}{\text{egg}} \times \text{\# eggs produced by patch i}} \\ 
%SP_i &= \text{LEP} \times S_e \times \frac{p_{i,i} \times \text{\# recruits from site $i$}}{S_e \times \text{\# eggs produced by patch $i$}} \\
%SP_i &= \text{LEP} \times S_e \times p_{i,i}. \label{EQN_SP}
SP_i &= \text{LRP} \times p_{i,i}. \label{EQN_SP}  % check this...
\end{split}
\end{equation}
A patch is self-persistent if $\text{SP} \geq 1$. If at least one patch is self-persistent, the metapopulation as a whole is persistent as well \citep{hastings_persistence_2006, burgess2014beyond}.

\paragraph*{Realized connectivity matrix and network persistence}
We find the probabilities of a recruit dispersing between each set of sites ($p_{i,j}$) by integrating the dispersal kernel (eqn.\ \ref{EQN_DispKernel}) over the distances between sites. We then create a realized connectivity matrix $C$ by multiplying the dispersal probabilities by the expected number of recruits an individual produces: $C_{i,j} = \text{LRP} \times p_{i,j}$ \citep[][though we include egg-recruit survival in LRP, rather than in $p_{i,j}$ as they do]{burgess2014beyond}. The diagonal entries of $C$, where the origin and destination are the same site, are the values of self-persistence we calculate above. 

Network persistence requires that the largest real eigenvalue of the realized connectivity matrix $\lambda_C$ be greater than $1$: $\text{NP} = \lambda_C > 1$ \citep[e.g.][]{hastings_persistence_2006, white_population_2010, burgess2014beyond}.

\paragraph*{Defining recruit and census stage} 

When assessing persistence, it is important to consider mortality and reproduction that occurs across the entire life cycle to determine whether an individual is replacing itself with an individual that reaches the same life stage \citep{burgess2014beyond}. We define a recruit to be a juvenile individual that has settled on the reef within the previous year, which also encompasses the size we are first able to sample (3.5-6.0 cm for parentage studies) (Fig.\ \ref{FIG_Schematic}, Life stage). In theory, it does not matter how we define recruit as long as we use that definition in our calculations of both egg-recruit survival and LEP. In our system, however, while it is straighforward to calculate LEP from any size, we do not have enough tagged recruits to reliably estimate survival to different recruit sizes. Instead, we choose the mean size of offspring matched in the parentage study as our best estimate of the size of a recruit ($\text{size}_\text{recruit}$) and test sensitivity to different recruit sizes by pulling from a uniform distribution over the sizes the recruit stage covers (3.5-6 cm, Table \ref{APP_TAB_Params}).

\begin{figure}[H] % Schematic
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/Schematic/Schematic.pdf}}
	\caption{Here, we show the data collected for fish at each life stage (life stage boxes are not scaled by length of stage) and how the empirical data fit into our parameter and metric calculations. \label{FIG_Schematic}} % MAKE CAPTION BETTER! MAKE SCHEMATIC BETTER TOO!
\end{figure}

\subsection*{Study system}

We focus on a tropical metapopulation of yellowtail clownfish (\textit{Ampiprion clarkii}, Fig.\ \ref{FIG_Map_and_photo}c) on the west coast of Leyte island facing the Camotes Sea in the Philippines (Fig.\ \ref{FIG_Map_and_photo}a). Like many clownfish species, yellowtail clownfish have a mutualistic relationship with anemones, where small colonies of fish live \citep{buston2003social, fautin1992field}. Yellowtail clownfish are protandrous hermaphrodites and maintain a size-strutured hierarchy; within an anemone, the largest fish is the breeding female, the next largest is the breeding male, and any smaller fish are non-breeding juveniles. The fish on an anemone maintain a strict social and size hierarchy \citep{buston2003social}, with fish moving up in rank to become breeders only after the larger fish have died. In the tropical patch reef habitat of the Philippines, yellowtail clownfish spawn once per lunar month from November to May, laying clutches of benthic eggs that the parents protect and tend \citep{ochi1989mating, holtswarth2017fecundity}. Larvae hatch after about six days and spend 7-10 days in the water column before returning to reef habitat to settle in an anemone \citep{fautin1992field}.

% % Other possible life histories in temperate areas or depending on the density of anemones
% \cite{hattori1991life}: different life history pathways found in population of \textit{A. clarkii} in Japan in area where they are the only anemonefish:
% \cite{ochi1989mating}: p.258: could be differences in the mating system/behavior of clownfish depending on how dense the host anemones are in the habitat and how costly it is to move between them; \textit{A. clarkii} in temperate areas off coast of Japan "showed many differences in behavior and morphology in comparison with conspecifics and othre anemonefishes from the primary habitats (coral reefs)."

Clownfish are particularly well-suited to metapopulation studies due to their limited movement as adults and clearly patchy habitat. Once fish have settled, they tend to stay within close proximity of their anemones \citep[XX meters,][]{stuartInPrephomerange}. This makes fish easier to relocate for mark-recapture studies and simiplifies the exchange between patches to only the dispersal during the larval phase. Patches, whether considered to be the reef patch or the anemone territory of the fish, are clearly discrete and easily delineated (Fig.\ \ref{FIG_Map_and_photo}a, b), which makes determining the spatial structure of the metapopulation clear. Additionally, clear patches make it easier to assess how much of the site has been surveyed. These simplifying characterstics in habitat and fish behavior make clownfish and other similarly territory-based reef fish useful model systems for studies of metapopulation dynamics and persistence \citep[e.g.][]{buston2013marine, salles_coral_2015, johnson2018integrating}. Our focal species of yellowtail clownfish tends to behave more like larger reef fishes, with territories that can extend beyond single anemones \citep{hattori1991life, ochi1989mating} and strong enough swimming skills that movement between patch reefs is possible though unusual \citep[seen XX times at our sites,][]{stuartInPrephomerange}, than the smaller clownfish \textit{A. percula} commonly used in previous metapopulation studies \citep[e.g.][]{buston2011probability, salles_coral_2015}.

\paragraph*{Field data collection}

We focus on a set of seventeen patch reef sites spanning approximately 30 km along the western coast of Leyte island (Fig.\ \ref{FIG_Map_and_photo}a). The sites consist of rocky patches of coral reef and are separated by sand flats (Fig.\ \ref{FIG_Map_and_photo}b). Previous work using genetic isolation by distance estimated that yellowtail clowfish larvae have a dispersal spread of about 10 km \citep[range 4-27 km,][]{pinsky2010using}, so our sites were selected to cover and exceed that range. On the north edge, the sites are isolated from nearby habitat with no substantial reef habitat for at least 20 km. % Pinsky et al. 2010 paper: (4–27 km with median 11 km)

\begin{figure}[H] % Map + clownfish photo
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/Map_and_photo.pdf}}
	\caption{a) Map of the sites along the coast of Leyte in the Philippines. b) Zoomed-in map of the two northern-most sites, Palanas and Wangag, to show anemone arrangement, with anemones occupied by \textit{A.\ clarkii} (green) or unoccupied by clownfish (orange). c) An example anemone occupied by \textit{A.\ clarkii} in a typical habitat at the sites. The metal anemone tag is visible just above the anemone on the rock.  \label{FIG_Map_and_photo}} 
\end{figure}

Since 2012, we have sampled fish and habitat at most of the sites annually (Table \ref{APP_TAB_PercHabSampled}). During sampling, divers using SCUBA and tethered to GPS readers swam the extent of each site. Divers visited each anemone inhabited by yellowtail clownfish, tagging the anemone to track it through time. At each anemone, the divers attempted to catch all of the yellowtail clownfish 3.5 cm and larger, taking a small tail fin-clip from each for use in genetic analysis, measuring the fork length, and noting the tail color (as an indicator of life stage). Starting in the 2015 field season, fish 6.0 cm and larger were also tagged with a passive integrated transponder (PIT) tag, unless already tagged. Divers also looked for eggs around each anemone and measured and photographed any clutches found. In total, we took fin clips from and genotyped 2772 fish and PIT-tagged 1929 fish across all years and sites combined, marking 3413 individual fish. % with an average of XX fish clipped and XX fish tagged per year.

%\paragraph*{Genotyping and parentage analysis}
\paragraph*{Parentage analysis and dispersal kernel}  % Check with Katrina/Michelle that this captures the highlights and includes the right references

We genotyped 2772 fish from our tissue samples and used single nucleotide polymorphisms to identify parent-offspring matches with the software program COLONY2 \citep{wang2012computationally} \citep[details on genotyping and parentage analysis in][]{catalanoInPrepconnectivity}. %2772 (# of gen_ids) doesn't match number of parents genotyped (1719) + # offspring genotyped (785) = 2504. Why not? %We digested extracted genomic DNA from our tissue samples using the ddRADseq protocol \citep{peterson2012double}, filtered the sequences with the bioinformatics pipeline dDocent \citep{puritz2014ddocent}, and called singled nucleotide polymorphisms (SNPs) using the program FreeBayes (\textit{is there a citation for this?}) \citep[details and results in][]{catalanoInPrepconnectivity}. 

Using the method described in \citep{bode2018estimating}, we fit a distance-based dispersal kernel \citep{catalanoInPrepconnectivity}, where the relative dispersal is a function of distance $d$ as measured in kilometers and parameters $\theta$ and $z = e^{k_d}$, which control the shape and scale of the kernel:
\begin{equation}
%p(d) = e^k e^{-(e^k d)^\theta}. \label{EQN_DispKernel}
p(d) = ze^{-(zd)^\theta}. \label{EQN_DispKernel}
\end{equation}
We use a Laplacian dispersal kernel with shape parameters $\theta = 1$ and scale parameter $k_d = -2.11$ (Fig.\ \ref{FIG_ParameterInputs}a, estimated in \citep{catalanoInPrepconnectivity}). To account for uncertainty in the dispersal kernel, we keep the shape parameter $\theta$ constant and pull the scale parameter $k_d$ from a set capturing the 95\% interval produced during kernel estimation in \cite{catalanoInPrepconnectivity}.

The dispersal kernel is estimated using fish that have already recruited to a population and survived to be sampled so it gives the relative amount of dispersal given that a fish recruits somewhere, not the probability that a released larva will travel a particular distance. To find the probability of fish dispersing among our sites, we numerically integrate the dispersal kernel (eqn.\ \ref{EQN_DispKernel}) using the distance from the middle of the origin site to the closest and farthest bounds of the destination site as the upper and lower bounds. For example, the probability of dispersal from site A to B, where $d_1$ is the distance from the middle of A to the closest edge of B and $d_2$ is the distance from the middle of A to the far edge of B, is:
\begin{equation} % might not need this equation...
%p_{A, B}(d) = \int_{d_1}^{d_2} e^k e^{-(e^k d)^\theta}  dd. \label{EQN_integratingDK}
p_{A, B}(d) = \int_{d_1}^{d_2} z e^{-(zd)^\theta}  dd. \label{EQN_integratingDK}
\end{equation}

\subsection*{Estimating inputs from empirical data} 

\paragraph*{Growth and survival: mark-recapture analyses}

We mark fish through both genetic samples and PIT tags, allowing us to estimate growth and survival through mark-recapture. After matching up recaptures of the same fish identified by genotype or tag, we have a set of encounters of 3413 marked fish that includes size and stage at each capture time.

For growth, we estimate the parameters of a von Bertalanffy growth curve \citep{fabens1965properties} in the growth increment form relating the length at first capture $L_t$ to the length at a later capture $L_{t+1}$ \citep{hart2009estimating}, where $L_\infty$ is the average asymptotic size across the population and $K$ controls the rate of growth: %check that I actually defined those well... % in the growth increment form (according to Hart and Chute) - need to actually get a copy of the Fabens paper, doesn't seem to be available on Google Scholar

\begin{equation} \label{EQN_VBL} 
\begin{split}
L_{t+1} & = L_t + (L_\infty - L_t)[1 - e^{(-K)}] \\
 & = e^{(-K)}L_t + L_\infty[1 - e^{(-K)}].
\end{split}
\end{equation}

We see from eqn.\ \ref{EQN_VBL} that we would expect the first length $L_t$ and the second length $L_{t+1}$ to be related linearly \citep{hart2009estimating}. From the slope $m = e^{(-K)}$ and y-intercept $b =  L_\infty[1 - e^{(-K)}]$, we can estimate the von Bertalanffy parameters, such that $K = -\ln m$ and $L_\infty = \frac{b}{(1-m)}$. We use the first and second capture lengths for fish that were recaught after a year (within 345 to 385 days) to estimate $L_\infty$ and $K$. We have some fish that were recaptured multiple times so we randomly select only one pair of recaptures from each to use in estimating the parameters, then repeat this process 1000 times to generate a distribution. 

We use the full set of marked fish to estimate annual survival $\phi$ and probability of recapture $p_r$ using the mark-recapture program MARK implemented in R \citep{RMark_Laake2013}. We consider several models with year, size, and site effects on the probability of survival on a log-odds scale (see full list in Table \ref{APP_TAB_MARKmodels}). For fish that are not recaptured in particular year, we estimate their size using our growth model (eqn.\ \ref{EQN_VBL}) and the size recorded or estimated in the previous year. Because fish are not well-mixed at our sites and instead stay quite close to their home anemones, we need to swim near an anemone to have a reasonable chance of capturing the fish on it. Therefore, we also consider a distance effect on recapture probability; we use the GPS tracks of divers to estimate the minimum distance between a diver and the anemone for each tagged fish in each sample year and include it as a factor in some of the models (Table \ref{APP_TAB_MARKmodels}). 

\paragraph*{Fecundity}

We use a size-dependent fecundity relationship, determined using photos of egg clutches and females \citep{yawdoszynInPrepfecundity}, where the number of eggs per clutch ($E_c$) is exponentially related to the length in cm of the female ($L$) with size effect $\beta_l = 2.388$, intercept $b = 1.174$, and egg age effect $\beta_e = -0.6083$ dependent on if the eggs are old enough to have visible eyes: 
\begin{equation} % is this the best way of writing this?
\ln(E_c) = \beta_l\ln(L) + \beta_e[\text{eyed}] + b. \label{EQN_Fec}
\end{equation}
To get total annual fecundity $f$, we multiply the number of eyed eggs per clutch by the number of clutches per year $c_e = 11.9$, using the estimate from \cite{holtswarth2017fecundity}.

We only consider reproductive effort once the fish has reached the female stage and use the average size of first observation as female for recaptured fish as the transition size $L_f = 9.32 \text{cm}$. To incorporate uncertainty, we draw from the full set of sizes at which recaptured fish were first captured as female (5.2 cm - 12.7 cm).

% For comparison of numbers (from Clownfish\_SP\_Notes:) Moyer (1986): the individual observed for 11 years and thought to live to at least 13 was ”estimated to have contributed about 160,000 propagules in its lifetime,” spent 3 years as a functional male, then outlived 3 mates as a female; fertilized about 45,000 eggs during 3 years as a male, then spawned about 115,000 eggs as a female; fecundity estimate at this site (Miyake-jima in Japan) is 17,500 eggs/yr/female (from Bell (1976))

\paragraph*{Lifetime egg production}
We use an integral projection model (IPM) \citep[e.g.][]{rees2014building} with size as the continuous structuring trait $z$ to estimate lifetime egg production (LEP), the total number of eggs produced by one individual, starting at the recruit stage. We initialize the IPM with one recruit-sized individual ($\text{size}_\text{recruit}$) at the initial time step ($t=0$), then project forward for 100 time steps using the size-dependent survival (eqn.\ \ref{EQN_Survival}) and growth (eqn.\ \ref{EQN_VBL}) functions as the probability density functions that make up the kernel to describe the survival and growth of the individual into the next time step. This gives us the size distribution at each time step, which represents the probability that the individual has survived and grown into each of the possible size categories, ranging from a minimum of $L=0$ cm to a maximum of $U=15$ cm. The probability that the individual is still alive and of any size decreases as the time steps progress; by using a large number of steps, we are able to avoid arbitrarily setting a maximum age and instead let the probabilities become essentially zero. 

We then multiply the size-distribution $v_z$ at each time by the size-dependent fecundity $f_z$ described above (eqn.\ \ref{EQN_Fec}) to get the total number of eggs produced at each time step. We then integrate across time and size to get the total number of eggs one individual is likely to produce in its lifetime:

\begin{equation}
\text{LEP} = \int_{t=0}^{100}\int_{z=L}^{z=U} v_{z,t} f_z dz dt. \label{EQN_LEP}
\end{equation}

To compute LEP, we discretize time and size and sum across the matrix. We use 0.1 as the standard deviation of size for a recruit ($\text{size}_{\text{recruit}, sd}$) and estimate the standard deviation of the distribution of sizes of fish in the next year ($\text{size}_{sd}$) from our recapture data (\ref{APP_TAB_Params}).
% We then multiply each size-distribution vector $v_z$ in the matrix by the size-dependent fecundity function described above (eqn.\ \ref{EQN_Fec}) to get the total number of eggs produced at each time step. To get the total number of eggs one individual is likely to produce in its lifetime, we then sum across all time steps in the individual's potential life.  

% % \begin{equation}
% % \text{LEP} = \int_{t=0}^{100}
% \begin{equation} % I'm not sure this is quite right...
% \text{LEP} = \Sigma_{t=0}^{t=100} \Sigma_{z=L}^{z=U} v_z,t f_z. \label{EQN_LEP}
% \end{equation}

\paragraph*{Survival from egg to recruit}

We estimate survival from egg to recruit ($S_e$) using parentage matches to estimate the number of surviving recruits produced by genotyped parents \citep[similar to the method in][]{johnson2018integrating}. We scale the number of offspring we match back to parents ($R_m = 62$) by various ways we could have missed offspring ($P_h$, $P_c$, $P_d$, and $P_s$, described below), then divide by the estimated number of eggs produced by genotyped parents, found by multiplying the number of genotyped parents ($N_g = 1719$) by the expected lifetime egg production for a fish of parent size ($\text{LEP}_p$):

\begin{equation}
S_e = \frac{\frac{R_m}{P_h P_c P_d P_s}}{N_g \text{LEP}_p}. \label{EQN_EggRecruitSurv}
\end{equation}

We scale the number of matched recruits we find by the cumulative proportion of habitat in our sites we sampled over time ($P_h = 0.41$, details in \ref{APP_SEC_ProbHabSampled}),
the probability of capturing a fish if we sampled its anemone ($P_c = 0.56$, see \ref{APP_SEC_ProbR} for details), and the proportion of the total dispersal kernel area from each our of sites covered within our sampling region ($P_d =0.57$, calculation in \ref{APP_SEC_PropDispKernelSampled}). Finally, because our dispersal kernel gives the probability of dispersal given that a recruit settled somewhere but our sampling region is not all habitat, we scale by the proportion habitat in our sampling region ($P_s = 0.20$, details in \ref{APP_SEC_PropHabInSampledRegion}) to avoid counting this mortality twice. 

To estimate $\text{LRP}_\text{local}$, we scale only by the proportion of habitat we cumulatively sample in our sites and the probability of capturing a fish.

To incorporate uncertainty in our estimate of egg-recruit survival, we consider uncertainty in the number of offspring assigned to parents during the parentage analysis ($R_m$) and in the probability of capturing a fish ($P_c$). We generate a set of values for the number of assigned offspring using a random binomial, where the number of trials is the number of genotyped offspring (745) and the probability of success on each trial is the assignment rate of offspring from the parentage analysis (0.079) \citep{catalanoInPrepconnectivity}. For the probability of capturing a fish, we pull values from a beta distribution that captures the mean and variance of capture probabilities across recapture dives (details in \ref{APP_EQN_ProbCapBetaDistParams}).

\paragraph*{Accounting for density-dependence}  % Should this be its own section or part of the survival from egg to recruit?

Ideally we would assess persistence metrics when the population is at low abundance and not limited by density-dependence. Clownfish have strong social hierarchies and juveniles on an anemone will prevent others from settling there as well \citep[seen in \textit{A. percula,}][]{buston2003forcible}. Each anenome, therefore, can only house one settling clownfish, with anemones already occupied by \textit{A. clarkii} settlers essentially unavailable as habitat. We attempt to account for this density-dependent mortality by multiplying our estimate of settling recruits (the numerator of eqn.\ \ref{EQN_EggRecruitSurv}) by the proportional increase ($\text{DD}$) in unoccupied anemones at our sites if all of the \textit{A. clarkii} anemones were unoccupied, where $p_A$ is the proportion of anemones occupied by \textit{A. clarkii} and $p_U$ is the proportion of unoccupied anemones: $\text{DD} = \frac{(p_U + p_A)}{p_U}$. We present results both with and without this density-dependence modification.

\paragraph*{Estimated abundance over time}

We also consider trends in abundance of breeding females at each site over time to compare to our replacement-based estimates of persistence. Similarly to as we do for offspring, we scale up the number of females caught at each site $i$ in each sampling year $t$ by the proportion of habitat sampled in that site and year $P_{h_{i,t}}$ and by the probability of capturing a fish $P_c$:
\begin{equation}
\text{\# females}_{i,t} = \frac{\text{\# females captured}_{i,t}}{P_{h_{i,t}}P_c}. \label{EQN_FemaleAbundance}
\end{equation}

We then fit a linear model through the time series for each site $i$ sampled in at least three years to assess whether the slope over time is positive or negative:  %indicates growth, decline, or stability in abundance:
\begin{equation}
%\text{\# females} = m*\text{year} + b. \label{EQN_AbundanceLM}  % There has got to be a better way to write this!
\text{\# females}_i \sim \text{year}
\end{equation}

\section*{Results}

Our estimated abundance of females at each site over time does not suggest a clear trend (Fig.\ \ref{FIG_FthroughTime}). In our very simple look at whether abundance seems to be increasing or decreasing, eleven sites had a positive slope over time and five had a negative slope (Fig.\ \ref{FIG_FthroughTime}q). For the two largest sites, with a mean estimated number of females of between 150-200, one has a positive slope (Wangag, Fig.\ \ref{FIG_FthroughTime}b) and one has a negative slope (Sitio Baybayon, Fig.\ \ref{FIG_FthroughTime}p) and the next two largest sites are also split (Palanas, Fig.\ \ref{FIG_FthroughTime}a and Haina, Fig.\ \ref{FIG_FthroughTime}o). Overall, there is not a clear directional change in abundance across the sites we sample over our sampling period.

\begin{figure}[H] %  abundance trends through time with some sort of time series analysis
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/Time_series_scaled_F_by_site_with_lines.pdf}}
	\caption{The estimated number of females at each site over the sampling years for sites sampled in at least three years. The total number of females at each site was estimated by scaling up the number of females captured at each site in each year by the proportion of habitat sampled at that site that season (see \ref{APP_SEC_ProbHabSampled} for details) and by the average probability of capturing a fish (see \ref{APP_SEC_ProbR}). We show the estimated abundances and trend for each site individually (a-p) and a histogram of the slopes of abundance through time (q). \label{FIG_FthroughTime}}
\end{figure}

From the mark-recapture analysis of tagged and genotyped fish, we estimate mean values of $L_\infty = 10.71 \text{cm}$ (range of estimates 10.50 - 10.90 cm) and $K = 0.864$ (range of estimates 0.785 - 0.944) for the von Bertalanffy growth curve parameters (eqn.\ \ref{EQN_VBL}, Fig.\ \ref{FIG_ParameterInputs}b, Table \ref{APP_TAB_Params}). For juvenile and adult (post-recruitment) survival on a log-odds scale, the best-fit model has an effect of size, with coefficient $b_a = 0.169 \pm 0.028$ SE and intercept $b_\phi = -1.83 \pm 0.231$ SE (eqn.\ \ref{EQN_Survival}). The accompanying best-fit model for log-odds recapture probability has a negative size effect and a negative effect of diver distance from the anemone (eqn.\ \ref{APP_EQN_MARKRecapture}, Fig.\ \ref{APP_FIG_RecapSizeDistEffect}).

\begin{equation}
\log(\frac{\phi}{1-\phi}) = b_\phi + b_a\text{size}. \label{EQN_Survival}
\end{equation}

\begin{figure}[H] % demographic parameters: survival curve, growth curve, dispersal kernel, transition size to female 
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/Parameter_inputs.pdf}}
	\caption{Best estimates (solid black line) and range included for uncertainty (gray) for dispersal (a), growth, including the 1:1 line in thick black (b), post-recruit survival (c), and size at female transition (d) parameters.  \label{FIG_ParameterInputs}}
\end{figure}

Using our best estimates for growth, survival, and fecundity, we calculate a value for LEP of 1061, ranging from 39 to 10345 when we consider uncertainty in the inputs (Fig.\ \ref{FIG_LEP_RperE_LRP}a). Adult survival has the most effect on the value of LEP (Fig.\ \ref{APP_FIG_Uncertainty_LEP}), with higher values of LEP the higher annual survival of adults. 

We estimate egg-recruit survival $S_e$ to be 7.8e-04, ranging from 1.2e-04 to 0.033 when we include uncertainty in the number of offspring assigned to parents and the probability of catching a fish (Fig.\ \ref{FIG_LEP_RperE_LRP}b). When we compensate for density-dependence in our data, we estimate $S_e$ to be 0.0013, ranging from 2.1e-04 to 0.057. These are somewhat high values of egg-recruit survival compared to what we see elsewhere in the literature \citep[e.g.][]{rumrill_natural_1990, metaxas_quantifying_2009} \citep[though not unreasonable, e.g.][]{white2014planktonic, johnson2018integrating} because we scale up by the amount of habitat in our sampling area and count mortality due to dispersal to non-habitat in the dispersal probability, rather than in $S_e$. Uncertainty in the size of transition to breeding female $L_f$ has the largest effect on egg-recruit survival (Fig.\ \ref{APP_FIG_Uncertainty_RperE}); we only consider reproduction from females, to avoid double-counting, so the larger the transition size to female, the fewer tagged eggs we estimate were produced by genotyped parents and the higher egg-recruit survival. % REPHRASE THIS BETTER! THEN TALK ABOUT LRP - what that means for persistence potential.

We estimate lifetime recruit production (LRP), the product of LEP and $S_e$, to be 0.83, with a range of 0.28 - 3.89 when we consider uncertainty in inputs. When we compensate for density-dependence, we estimate a value of 1.42 for LRP, with a range of 0.48 - 6.66. The value when we compensate for density-dependence and the range of uncertainty for both are above the threshold of one necessary for replacement before considering dispersal. This mean that individuals at our sites produce enough surviving offspring before considering dispersal to be able to replace themselves, but LRP does not tell us whether those offspring will settle within our sample sites and drive persistence. %This suggests that even without considering connectivity, the individuals at our sample populations do not produce enough offspring that survive to recruitment to replace themselves. When we consider uncertainty in our parameter estimates, we do see a few cases where LRP $>$ 1, but the majority are well below the threshold for replacement.

We also estimate replacement for recruits from our sites returning to our sites,  $\text{LRP}_\text{local}$, which implicitly includes dispersal mortality, to be 0.09 (ranging from 0.03 to 0.44 when we include uncertainty) or 0.16 (0.05 to 0.76) when we compensate for density-dependence. With a value well below one, this suggests individuals at our sites do not replace themselves with recruits that settle in our sites, suggesting our sites do not persist as an independent network. When we calculate $\text{LRP}_\text{local}$ using all arriving recruits to our sites, however, rather than just those originating there, the best estimates are $> 1$ whether or not we compensate for density dependence (2.06, 1.22, respectively), suggesting that there is recruit-recruit replacement at our sites when we include immigrant recruits.

\begin{figure}[H] % LEP, recruits-per-egg, LRP shown with uncertainty, line for best estimate (using mean offspring size as recruit size)
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/LEP_RperE_LRP.pdf}}
	\caption{Estimates of a) LEP, b) egg-recruit survival, and c) LRP, and d) $\text{LRP}_\text{local}$, showing the best estimate (black solid line) and range of estimates considering uncertainty in the inputs. The estimates in e-g include our attempt to remove density-dependence in the early life stages, while the estimates in b-d do not. We only consider density-dependence in egg-recruit survival so LEP (a) estimates are the same in both cases. \label{FIG_LEP_RperE_LRP}}
\end{figure}

We do not find any sites with SP $>$ 1, whether we compensate for density-dependence or not (Fig.\ \ref{FIG_SP}), indicating that no site could persist in isolation. Given that our best estimate of LRP does not suggest replacement and only a fraction of those offspring stay at the natal site, this makes sense. We see the highest values of self-persistence at Haina ($\text{SP} = 0.079$, $0.13$ when compensating for density-dependence) and Wangag ($\text{SP} = 0.048$, $0.082$ when compensating for density-dependence), our two widest sites. 

\begin{figure}[H] % SP with line for best estimate (using mean offspring size as recruit size)
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/SP_hists_by_site_noSLSTCP.pdf}}
	\caption{Values of self-persistence at each site, showing the best estimate (black point) and range of estimates considering uncertainty in the input paramters. No site reaches the value SP $\geq$ 1 necessary to be self-persistent. The estimates in (b) attempt to compensate for density-dependence in early life stages in our data, while the estimates in (a) do not. \label{FIG_SP}}
\end{figure}

For network persistence, our best estimate of the dominant eigenvalue of the realized connectivity matrix $\lambda_c$ is 0.21 with a range of 0.07 - 0.92 (Fig.\ \ref{FIG_NP_realizedCmat}a), or 0.36 with a range 0.12 - 1.58 when we compensate for density-dependence (Fig.\ \ref{FIG_NP_realizedCmat}c). Our sites are likely not network persistent, as our best estimates and most of the values we see in our runs with uncertainty are below one, but network persistence is possible, as our range of estimates does exceed one when we compensate for density-dependence. We see that most of the connectivity occurs among the sites in the northern part of our sample area, from Palanas to Caridad Cemetery, and at the southern part of our sample area from Tomakin Dako to Sitio Baybayon (Fig.\ \ref{FIG_NP_realizedCmat}b, d), where the largest sites are. 

\begin{figure}[H] % NP with line for best estimate (using mean offspring size as recruit size), realized connectivity matrix for best estimate - should re-do without Sitio Lonas, Sitio Tugas, Caridad Proper!
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/NP_and_connMatrixR.pdf}}
	\caption{a) Network persistence values, showing the best estimate (black solid line) and range of estimates considering uncertainty, without accounting for density-dependence in our data. b) The realized connectivity matrix $C$, with sites arranged from north (Palanas) to south (Sitio Baybayon), without accounting for density-dependence in our data. Plots c) and d) show the network persistence values and realized connectivity matrix with attempts to compensate for density-dependence in early life stages in our data. \label{FIG_NP_realizedCmat}}
\end{figure}

Based on our estimates of LRP, $\text{LRP}_\text{local}$, SP, and NP, it is possible but not likely that our set of sites is able to persist in isolation as a closed system. With our site configuration and dispersal kernel estimate, we would need a value of LRP of 3.99 (an egg-recruit survival of 0.0038 with our estimated value of LEP or a value of LEP of 5095 or 2975 with our estimated egg-recruit survival compensating and not compensating for density-dependence, respectively), to have a best estimate of $\lambda_c = 1$ and network persistence. 

\section*{Discussion}

We do not see strong evidence for persistence in our metric estimates. We see no evidence for self-persistence where an individual site could persist alone (Fig.\ \ref{FIG_SP}) and weak evidence for network persistence; it is possible at the upper end of our range of estimates with uncertainty but not suggested by most of the range or our best estimates (Fig.\ \ref{FIG_NP_realizedCmat}). The abundances through time at our sites do not show a clear directional change, however, suggesting that the population at our sites is relatively constant but relies on input of recruits from outside sites to persist. The portion of coastline we sampled is likely a sink portion of a larger metapopulation. 

For our sites to be able to persist as a network on their own, the number of surviving recruits produced by an average recruit - LRP - would likely need to be higher. With our estimated connectivity, LRP would need to be at least 3.99 to see network persistence among our sites, which is within the top of our range of uncertainty but about 3-5 times higher than our best estimates, without and with density-dependence compensation. Our best estimate of LRP when we compensate for density-dependence is greater than one, so higher connectivity and retention of offspring among our sites could lead to network persistence, but almost all surviving offspring would need to be retained. At our best estimate without density-dependence compensation, however, LRP is less than one - the average recruit only produces 0.83 of a surviving recruit of the same stage - so no amount of increased retention or connectivity, even retaining all of the recruits produced from our sites, would lead to network persistence. Similarly, if other surrounding patch populations had a similar LRP, increasing the area of the network to include them would also not achieve network persistence. If nearby sites have higher egg production or survival to recruit, however, it might not take much of an increase in area considered to create a persistence network. Nearby reef sites such as Cuatro Islas have higher quality habitat and could be contributing recruits to our sites.   

% reconcile with dispersal distance
We do not find clear evidence for network persistence for our sites despite estimates of the mean dispersal distance of \textit{A. clarkii} from previous genetic work \citep[11 km,][]{pinsky2010using} and from our samples \citep[8.25 km, with 95\% confidence interval 7.41 to 9.36,][]{catalanoInPrepconnectivity} that are well within the 30 km span of our sites. Though the width of our sampling region is more than twice the mean dispersal distance, which \cite{lockwood2002effects} find sufficient for persistence of an isolated reserve, their estimate assumes continuous habitat within the reserve and our region is only about 20\% habitat. For a habitat configuration more similar to our system, habitat patches (reserves) spaced on a coastline with non-habitat in between, they find that either 40\% of the coastline needs to be preserved or a minimum patch size must be 1.25 times the mean dispersal distance to ensure persistence. Our largest site, Haina, is only about 0.8 km wide, about 10 times less than the mean dispersal distance, so it is possible we do not have enough habitat in our region for network persistence, exacerbated by our 4 largest sites being at the edges of our area and sending half of their recruits away from our sites. Our low, and possibly below-replacement, estimate for LRP also suggests that lack of persistence in these sites is not due to excessive dispersal out of the area but due to low production and survival of offspring. The reef health and habitat quality in our sites is generally low, due anthropogenic effects such as pollution and silt from a nearby gravel mine, and habitat disturbance due to storms. Our sites are in an area that was hit in 2013 by Typhoon Haiyan, one of the strongest typhoons ever to make landfall, which destroyed much of the reef habitat in some of our northern sampling areas. This recent disturbance and generally low habitat quality could contribute to low production of surviving recruits in our sites \citep[seen in other populations with low habitat quality, e.g.][]{hayashi2019low} necessitating subsidization by outside populations.

% uncertainty
We see considerable uncertainty in our estimate of persistence metrics depending on the particular input values we use (Figs.\ \ref{FIG_LEP_RperE_LRP}, \ref{APP_FIG_Uncertainty_NP}). Our highest estimate for LRP is about 24 times more than our lowest estimate and our highest NP estimate is about 22 larger than our lowest, spanning the range between network persistence for our set of sites to far from it. Measuring demographic and dispersal parameters in the field is challenging; in the face of limited and imperfect data, characterizing uncertainty and propagating it from our estimates of demographic and dispersal inputs through to our estimates of persistence metrics is important to contextualize our results. In our study, uncertainty in egg-recruit survival \citep[a commonly challenging parameter to estimate, e.g.][]{johnson2018integrating,hameed2016inverse}, partially driven by uncertainty in how likely we are to capture recruits during sampling (Figs.\ \ref{APP_FIG_Uncertainty_RperE}, \ref{APP_FIG_Uncertainty_RperE_DD}), has a large effect on whether or not we think our populations are persistent. For a marine metapopulation, our system is relatively uncomplicated and yet still hard in which to concretely assertain persistence. As we accumulate more empirical assessments of metapopulations to compare to our expectations from theory and models, we will have to think carefully about how to handle uncertainty as we move to tackling larger and more complicated systems. %Examples of when the empirical result doesn't match the model expectation?

% Caveats - density-dependence, could also discuss site-specific demographic parameters?
Persistence criteria, such as those detailed in \cite{hastings_persistence_2006} and \cite{burgess2014beyond}, ask whether a population at low abundance can grow and recover rather than going extinct. Density-dependence is often ignored at low abundances \citep[e.g.][]{caswell_matrix_2001, hastings_simple_2006} so is not explicitly considered in persistence metrics. In real populations, however, it can be challenging to estimate density-independent demographic rates, as density-dependence is occurring in the population as it is sampled. In \textit{A. clarkii}, density-dependence is likely most important in early life stages, as for many fish species, but could play an important role throughout the life history due to the social hierarchies in colonies of clownfish \citep[e.g.][]{buston2011determinants}. In other species of clownfish, individuals on the same anemone maintain strict size spacing, restricting their food intake and growth to avoid encroaching on the position of another fish and being attacked or evicted \citep[seen in \textit{A. percula},][]{buston2003forcible, buston2003social}. This suggests that while fish are in the pre-reproductive queue, density-dependence may lower growth rates compared to the growth of fish alone on an anemone, as would be the case in a population at low abundance. We attempt to account for the primary effect of density-dependence on our estimate of egg-recruit survival but other estimates, particularly growth and survival, would also likely be higher in the absence of density-dependence and increase LRP.

% Paragraph 6: Wrap-up
Our estimates of persistence metrics suggest that it is possible but not likely that the region of sites we sample persist as a network without outside input, despite covering an area more than twice the estimated mean dispersal distance for our focal species. Our estimate of LRP near the threshold of one required for replacement (slightly $ < 1$ when we do not compensate for density-dependence, slightly $ > 1$ when we do), suggests that dispersal is not likely the primary reason our sites do not persist as a network. If density-dependence is strongly present in our data such that our compensated estimate is the best, then our sites could persist if there were no losses to dispersal. Otherwise, our sites do not produce enough offspring for replacement regardless of dispersal patterns, possibly due to worsening habitat quality. This is a reminder that dispersal is only part of the persistence story for metapopulations; even areas that seem large enough to contain a persistent network based on dispersal distance will not be able to persist in isolation if they have low production and survival of offspring. We do find recruits coming back to our region, and even to their natal site, but broader connectivity to more productive sites likely enables our sites to persist.



% Our estimates of survival probabilities are similar to those estimated for other species of clownfish, particulary our relationship with size where small fish have a low annual survival and the largest fish have a high annual survival (CITATIONS, Buston paper, also compare to Salles et al. 2015). Our fecundity estimates are lower than those for \textit{A. clarkii} in temperate areas, almost XX times lower (CITATIONS, Ochi papers - 17,500 eggs/yr/female, from Bell 1976).

% Density-dependence notes: Settling clownfish can be ”driven from the anemone within hours of their arrival [38,47]” - p.1884 in Buston et al. (2011), citations are Buston (2003a) and Elliott et al. (1995)

% Context for dispersal kernel:
% \begin{itemize}
% 	\item previous research suggests 10km dispersal kernel spread for yellowtail clownfish \citep{pinsky2010using}
% 	\item \cite{hameed2016inverse}: \textit{Petrolisthes cinctipes}, porcelin crab, in N CA had mean dispersal distance of 6.9km (+-sd of 25km) despite 4-6 week PLD
% \end{itemize}

% Context for larval mortality or egg-recruit mortality:
% \begin{itemize}
% 	\item check \cite{white2014planktonic}
% 	\item check estimate in \cite{johnson2018integrating}
% \end{itemize}

% Context for adult mortality
% \begin{itemize}
% 	\item \cite{salles_coral_2015}: check paper, possibly 0.18-0.49 for J, 0.09-0.44 for M, 0.19-0.55 for F (are the ranges for subpops? biannual mortality rates?), orange clownfish (\textit{Ampiphrion percula}) in Kimbe Bay
% 	\item \cite{buston2003social} (or possibly the other 2003 Buston paper...), 12.9\% mortality per year for \textit{Ampiprion percula} in Papua New Guinea
% \end{itemize}


\newpage{}

{\LARGE Appendix}

\appendix

\renewcommand{\theequation}{A\arabic{equation}}
% redefine the command that creates the equation number.
\renewcommand{\thetable}{A\arabic{table}}
\setcounter{equation}{0}  % reset counter 
\setcounter{figure}{0}
\setcounter{table}{0}
\numberwithin{equation}{section}
\numberwithin{figure}{section}

\section*{Summary of parameters}

%[\textit{Need to clarify somewhere what kind of distributions are going into the uncertainty runs (drawn from data, uniform across a range, 95\% confidence bounds, etc.)}]
% Make table prettier
\begin{centering}
\begin{longtable}{|p{0.8in}|p{1.2in}|p{1.5in}|p{1in}|p{1.5in}|}
\hline 
\textbf{Parameter} & \textbf{Description} & \textbf{Best estimate} & \textbf{Range in uncertainty runs} & \textbf{Notes} \\ \hline
$k_d$ & scale parameter in dispersal kernel & -2.11 & -2.36 to -1.96 & estimated using methods in \cite{bode2018estimating} in Catalano et al.\ (in prep) \\ \hline
$\theta$ & shape parameter in dispersal kernel & 1 & NA & estimated using methods in \cite{bode2018estimating} in Catalano et al.\ (in prep) \\ \hline
$L_\infty$ & average asymptotic size in von Bertalanffy growth curve & 10.70 cm & 10.50 to 10.90 cm &  \\ \hline
$K$ & growth coefficient in von Bertalanffy growth curve &  0.864 & 0.785 to 0.944 & \\ \hline  
$b_\phi$ & intercept for adult survival & -1.82 & $\pm$ 0.231 standard error & on a log-odds scale \\ \hline
$b_a$ & size effect for adult survival & 0.169 & $\pm$ 0.028 standard error & on a log-odds scale \\ \hline
$b_{p_r}$ & intercept for recapture probability from mark-recapture analysis & 2.10 & $\pm$ 0.849 standard error & on a log-odds scale, not used in persistence estimates \\ \hline
$b_1$ & size effect for recapture & -0.161 & $\pm$ 0.088 standard error & on a log-odds scale, not used in persistence estimates \\ \hline
$b_2$ & distance effect for recapture & -0.196 & $\pm$ 0.023 standard error & on a log-odds scale, not used in persistence estimates \\ \hline
$\text{size}_\text{recruit}$ & size (cm) of recruited offspring & mean of size of offspring in parentage analysis = 4.37 cm & 3.5 - 6.0 cm & drawn from uniform distribution across range \\ \hline
$\text{size}_{\text{recruit}, sd}$ & standard deviation of size of a recruit & 0.1 & & used in discretization of IPM for LEP \\ \hline
$\text{size}_{sd}$ & standard deviation distribution of sizes of a fish in the next year & 1.45 & & used in discretization of IPM for LEP, estimated from range of sizes for fish starting at 7.4-7.6 cm and recaptured a year later \\ \hline
%$S_e$ & egg-recruit survival & & &  \\ \hline
%$E_c$ & eggs per clutch & depends on female size (eqn.\ \ref{EQN_Fec}) & & relationship from Yawdoszyn et al.\ (in prep) \\ \hline
$b_e$ & coefficient for eyed eggs & -0.608 & & Yawdoszyn et al.\ (in prep) \\ \hline
$b_l$ & size effect in eggs-per-clutch relationship & 2.39 & & Yawdoszyn et al.\ (in prep) \\ \hline
$b$ & intercept in eggs-per-clutch relationship & 1.17 & & Yawdoszyn et al.\ (in prep) \\ \hline
$c_e$ & egg clutches per year & 11.9 & & \cite{holtswarth2017fecundity} \\ \hline
$L_f$ & size at transition to female & 9.32cm & 5.2 - 12.7cm & drawn from distribution in data \\ \hline
$P_h$ & proportion of sites sampled cumulatively across time & 0.41 & & details in \ref{APP_SEC_ProbHabSampled} \\ \hline
$P_d$ & proportion of dispersal kernel area from each site covered by our sampling region & 0.57 & & details in \ref{APP_SEC_PropDispKernelSampled} \\ \hline 
$P_c$ & probability of capturing a fish & 0.56 & drawn from beta distribution with parameters $\alpha_{P_c} = 1.44$ and $\beta_{P_c} = 1.13$ & details in \ref{APP_SEC_ProbR} \\ \hline
$P_s$ & proportion of our sampling region that is habitat & 0.20 & & details in \ref{APP_SEC_PropHabInSampledRegion} \\ \hline
$\text{DD}$ & proportion of habitat that would be available without density-dependence at settlement & 1.71 & & \\ \hline
$p_U$ & proportion of anemones unoccupied by clownfish & 0.53 & & used to estimate DD \\ \hline
$p_A$ & proportion of anemones occupied by \textit{A. clarkii} & 0.38& & used to estimate DD \\ \hline
\caption{}\label{APP_TAB_Params}
\end{longtable}
\end{centering}

\section{Method details}

\subsection{Proportion of habitat sampled} \label{APP_SEC_ProbHabSampled}

We used tagged anemones to estimate the proportion of habitat sampled at each site in each year ($P_{h_{i,t}}$). We tagged each anemone that is home to \textit{A. clarkii}, with a metal tag, which is relatively permanent and easy to re-sight (the anemone tag is visible above the anemone in Fig.\ \ref{FIG_Map_and_photo}c), so we consider the total number of metal tags at each site to be the total number of anemones that are habitat. We divide the number of tagged anemones visited each sampling year by the total number of metal tags at that site to get the proportion of habitat sampled. We use proportion of anemones rather than proportion of total site area because anemones, and therefore habitat quality, are unevenly distributed across the site; areas we did not visit are likely to have a lower density of anemones than the areas we did sample.

For scaling the number of tagged recruited offspring to account for areas of our sites we did not sample, we use the overall proportion habitat sampled across all sites and sampling years ($P_h$). We sum the metal-tagged anemones we visited across all sites and years to get the total number of metal-tagged anemones we visited while sampling. We then divide that by the number of anemones we could have sampled, the sum of total metal-tagged anemones across all sites multiplied by the number of sampling years, to get the overall proportion habitat sampled across our sites and sampling years.

%\textit{Add details about how sometimes it is >1 if the site doesn't have metal tags? Mention plastic tags?}

\begin{table}
\begin{centering}
\begin{tabular}{|l|r|r|r|r|r|r|r|r|}
\hline 
\multicolumn{2}{|c|}{} & \multicolumn{7}{|c|}{\% Habitat surveyed} \\ \hline
Site & \# Total anems & 2012 & 2013 & 2014 & 2015 & 2016 & 2017 & 2018 \\ \hline
Cabatoan & 26 & 42 & 58 & 58 & 65 & 73 & 0 & 62 \\ \hline
Caridad Cemetery & 4 & 0 & 75 & 50 & 0 & 50 & 50 & 50 \\ \hline
Elementary School & 8 & 0 & 100 & 38 & 88 & 88 & 88 & 100 \\ \hline
Gabas & 9 & 0 & 0 & 0 & 44 & 44 & 67 & 0 \\ \hline
Haina & 104 & 0 & 6 & 13 & 13 & 10 & 56 & 80 \\ \hline
Hicgop South & 18 & 0 & 67 & 22 & 28 & 56 & 83 & 78 \\ \hline
N. Magbangon & 105 & 5 & 12 & 40 & 63 & 63 & 0 & 5 \\ \hline
S. Magbangon & 34 & 41 & 56 & 32 & 0 & 65 & 0 & 71 \\ \hline
Palanas & 137 & 29 & 58 & 47 & 63 & 85 & 86 & 86 \\ \hline
Poroc Rose & 13 & 100 & 100 & 69 & 31 & 23 & 69 & 69 \\ \hline
Poroc San Flower & 11 & 100 & 82 & 73 & 73 & 55 & 82 & 64 \\ \hline
San Agustin & 17 & 94 & 65 & 71 & 65 & 100 & 82 & 76 \\ \hline
Sitio Baybaon & 260 & 0 & 14 & 30 & 33 & 30 & 41 & 80 \\ \hline
Tomakin Dako & 50 & 0 & 24 & 22 & 36 & 34 & 60 & 68 \\ \hline
Visca & 13 & 100 & 100 & 23 & 38 & 62 & 85 & 62 \\ \hline
Wangag & 296 & 18 & 32 & 42 & 34 & 26 & 49 & 68 \\ \hline
\end{tabular}
\end{centering}
\caption{Table showing the percent of metal-tagged anemones surveyed at each site in each sampling year.}\label{APP_TAB_PercHabSampled}
\end{table}

\newpage{}

\subsection{Probability of capturing a fish, from recapture dives} \label{APP_SEC_ProbR}

We use mark-recapture data from recapture dives done within a sampling season to estimate the probability of capturing a fish. During some of the sampling years (XX), portions of the sites were sampled again XX-XX weeks after the original sampling dives. We assume there is no mortality of tagged fish between the original sampling dives and the recapture dives because they are so close in time and that fish do not change their behavior or reponse to divers, so therefore assume that the probability of recapturing a fish is the same as the probablity of capturing a fish on a sample dive. For each recapture dive, we use GPS tracks of the divers to identify the anemones covered in the recapture dive and the set of PIT-tagged fish encountered on those anemones during the original sampling dives. We estimate the probability of capture $P_c$ as the number of tagged fish caught during the capture dive $m_2$ divided by the total number of fish caught on the recapture dive $n_2$: $P_c = \frac{m_2}{n_2}$. 

We use the mean $P_c$ across all 14 recapture dives, covering XX sites in 3 sampling seasons (2016, 2017, 2018), as our best estimate. Because there are so few recapture dives compared to the number of times we calculate the metrics to show the range of uncertainty, we represent the probability of capture as a distribution, rather than pulling directly from the values calculated for each recapture dive. The distribution of capture probabilities across the 14 dives is quite skewed so we represent it as a beta distribution, using the mean $\mu_{P_c}$ and variance $V_{P_c}$ of the set of 14 values to find the appropriate $\alpha_{P_c}$ and $\beta_{P_c}$ parameters, where 

\begin{eqnarray}
\alpha_{P_c} &=& (\frac{1-\mu_{P_c}}{V_{P_c}} - \frac{1}{\mu_{P_c}}) \mu_{P_c}^2 \\
\beta_{P_c} &=& \alpha_{\mu_{P_c}} \times \frac{1}{\mu_{P_c} - 1}. \label{APP_EQN_ProbCapBetaDistParams}  % Should I cite a source for this?
\end{eqnarray}

% Should I include a table of the recapture probs for each recapture dive, including site and year?

The mean of the individual capture probability values is $\mu_{P_c} = 0.56$, with variance $V_{P_c} = 0.069$, which gives beta distribution parameters $\alpha_{P_c} = 1.44$ and $\beta_{P_c} = 1.13$. We sample 1000 values from the beta distribution, then truncate the sample to only values larger than the lowest value of $P_c$ estimated in an individual dive (0.20), to avoid extremely low values that are sometimes sampled but are unrealistically low. We then sample with replacement from the truncated set to get a vector of values the length of the number of runs.

\subsection*{Proportion of dispersal kernel area sampled} \label{APP_SEC_PropDispKernelSampled}

[\textit{Add in description of calculation and equation}]

\begin{figure}[H] % Proportion of dispersal kernel area from each site sampled
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/Prop_of_kernel_area_sampled_by_site.pdf}}
	\caption{Proportion of the dispersal kernel area from the center of each site covered by our sampling. \label{APP_FIG_PropDispKernelAreaSampled}}
\end{figure}

\subsection{Proportion habitat in sampling area} \label{APP_SEC_PropHabInSampledRegion}

We assume that larvae are unable to navigate to habitat if they attempt to settle on an unsuitable patch, though clownfish larvae do likely have some ability both to sense habitat (CITATIONS) and move toward it (CITATIONS)). To avoid counting mortality due to settling on non-habitat twice - once in scaling up our matched recruits, which only includes those who settled on habitat, and once in integrating the dispersal kernel, we scale our estimate of total surviving recruits from our patches by the proportion of our sampling region that is habitat ($P_s$). We find $P_s$ by summing the lengths of all of our sites, which run approximately north-south, and dividing that by the total distance north-south of our sampling region, giving $P_s = 0.20.$

\newpage{}

\subsection{Full set of MARK models} \label{APP_MARKModels}
We consider the following set of models in MARK for survival ($\phi$) and recapture ($p$) probability, including effects of size ($S$), minimum distance from diver to anemone during surveys ($D$), time ($t$), and site ($i$) (Table \ref{APP_TAB_MARKmodels}):

\begin{table}
\begin{centering}
\begin{tabular}{|p{2in}|p{2.5in}|p{0.75in}|p{0.75in}|}
\hline 
\textbf{Model} & \textbf{Model description} & \textbf{AICc} & \textbf{dAICc} \\ \hline
$\phi \sim S$, $p \sim S+D$ & survival size, recapture size+distance & 3348.861 & 0 \\ \hline
$\phi \sim S$, $p \sim D$ & survival size, recapture distance & 3359.998 & -11.1371 \\ \hline
$\phi$, $p \sim D$ & survival constant, recapture distance & 3383.175 & 34.3141 \\ \hline
$\phi$, $p \sim S+D$ & survival constant, recapture size+distance & 3384.959 & 36.0981 \\ \hline
$\phi \sim t$, $p$ & survival time, recapture constant & 3408.342 & 59.4816 \\ \hline
$\phi \sim i$, $p$ & survival site, recapture constant & 3440.842 & 91.98112 \\ \hline
$\phi \sim i$, $p \sim S+D$ & survival site, recapture size+distance & 3440.842 & 91.98112 \\ \hline
$\phi$, $p \sim t$ & survival constant, recapture time & 3453.609 & 104.74839 \\ \hline
$\phi \sim S$, $p \sim S$ & survival size, recapture size & 3527.710 & 178.84940 \\ \hline
$\phi$, $p$ & survival constant, recapture constant & 3570.908 & 222.04690 \\ \hline
\end{tabular}
\end{centering}
\caption{}\label{APP_TAB_MARKmodels}
\end{table}

\paragraph{Recapture model} 
The best model for log-odds recapture probability, accompanying the survival model in eqn.\ \ref{EQN_Survival}, has a size effect ($b_1 = -1.816 \pm 0.080$ SE, Fig.\ \ref{APP_FIG_RecapSizeDistEffect}a) and a negative effect of diver distance from the anemone ($b_2 = -0.171 \pm 0.021$ SE, Fig.\ \ref{APP_FIG_RecapSizeDistEffect}b), with intercept $b_{p_r} = 17.93 \pm 0.858$ SE:

\begin{equation}
\log(\frac{p_r}{1-p_r}) = b_{p_r} + b_1\text{size} + b_2d. \label{APP_EQN_MARKRecapture}
\end{equation}
% \begin{eqnarray}
% \log(\frac{\phi}{1-\phi}) &=& b_\phi + b_a\text{size} \\
% \log(\frac{p_r}{1-p_r}) &=& b_{p_r} + b_1\text{size} + b_2d. \label{EQN_Survival}
% \end{eqnarray}
%These results suggest that larger fish have higher annual survival, which is similar to survival estimates in other clownfish species (check Buston paper). 
The negative effect of both size and distance suggest that divers are less likely to recapture larger fish and those at anemones far from areas sampled. %, with the chance of recapturing an average-sized fish falling below 5\% if a diver stays farther than XX from its home anemone.

% Add plot of distances to anems in different years?
% Add plot of effect of size and distance on recapture probability?
\begin{figure}[H] % 4 relationships among parameters
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/Recapture_size_distance_effects.pdf}}
	\caption{Effects of a) fish size and b) minimum distance between divers and the anemone where the fish was first caught on the probability the fish will be recaptured, estimated along with survival in a mark-recapture analysis. Size has a slightly negative effect on the probability of recapture - larger fish are stronger swimmers and more likely to flee the anemone when divers approach. Distance also has a negative effect on recapture, as fish tend to stay close to their anemones and are not likely to be recaught if divers do not get close the their anemone. \label{APP_FIG_RecapSizeDistEffect}}
\end{figure}

\section{Uncertainty and sensitivity}

% \subsection{What-if analyses}

% To compare our replacement-based persistence results, which do not suggest that our sites make up a persistent metapopulation, with our abundance trends (Fig.\ \ref{FIG_FthroughTime}, which suggest that population abundances at our site have been relatively stable over our sampling period, we estimate recruits arriving at our sites per recruit there, regardless of the origin of the arriving recruits. We repeat our metric estimates but use all offspring genotyped at our sites, scaled by proportion habitat sampled within our sites $P_h$ and the probability of capturing a fish $P_c$, as our estimate of recruited tagged offspring. We find a value of $\text{LRP}_\text{local} = 1.22$ without compensating for density dependence and 2.06 if we do, both of which are $> 1$, suggesting our sites have persistent populations, though cannot necessarily persist alone, as our NP values are still $>1$.

% \paragraph{All genotyped offspring at our sites originated from our sites}

% \begin{figure}[H] % Do we see replacement if we consider all recruits when looking at recuits-per-recruit?
% 	\centering
% 	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/PersistenceMetrics/Whatifs/LEP_R_and_NP_histograms_whatif_all_offspring.pdf}}
% 	\caption{a) Recruits per recruit when we consider all arriving recruits to have originated from our sites. b) Range of values of NP considering all arriving recruits to be offspring from our sites, with the best estimate in a black solid line. \label{APP_FIG_AllOffspringWhatIf}}
% \end{figure}

\subsection{Sensitivity to parameters}

% Range of parameters used as input for uncertainty runs

\begin{figure}[H] % Range of parameter inputs for uncertainty runs
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/Uncertainty_inputs.pdf}}
	\caption{Range of parameter inputs for uncertainty runs with all uncertainty included: a) $\text{size}_\text{recruit}$, the census size at which fish are considered to have recruited after egg-recruit survival occurs; b) $L_f$, the size at which fish transition from male to female and their reproductive output is included in the estimate of lifetime egg production (LEP); c) $k_d$, the scale parameter in the dispersal kernel; d) the parameters $L\infty$ and $K$ of the von Bertalanffy growth model; e) the intercept $b_\phi$ of the adult size-dependent survival relationship; f) $P_c$, the probability of capturing a fish; g) number of offspring assigned back to parents in the parentage analysis; h) factors that scale the number of estimated recruits from our site based on density-dependence in settler success (DD), proportion of the dispersal kernel captured by our sampling region ($P_d$), the cumulative proportion of our sites we sampled over time ($P_h$), and the proportion of our sampling area that is habitat ($P_s$). \label{APP_FIG_UncertaintyInputs}}
\end{figure}

% Relationships among parameters

\begin{figure}[H] % 4 relationships among parameters
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/Param_metric_relationships.pdf}}
	\caption{Relationships among parameters and metrics. a) We only count reproductive effort by fish in the female stage so the higher the transition size to breeding female, the fewer eggs parents are considered to produce, which increases the estiamted egg-recruit survival. b) LRP strongly affects NP by changing the number of potential recruits dispered through the connectivity matrix. c) The probability of capturing a fish does not have a clear relationship to NP. d) LEP is higher with higher survival estimates because fish are more likely to survive longer as reproducing adults. e) The size we consider to be a recruit marks the transition of mortality included in egg-recruit survival to mortality being captured by annual adult survival. Because we do not have the data to change egg-recruit survival to account for different recruit sizes, increasing the recruit size increases LRP by wrapping more mortality into the egg-recruit survival estimate, rather than LEP. \label{APP_FIG_ParamMetricRelationships}}
\end{figure} % Think about a better way to explain (e) and whether we should even include it.

\subsection{Effects of different types of uncertainty on metrics}

\paragraph{Lifetime egg production (LEP)}

Annual survival post-recruitment provides drives most of the uncertainty in LEP, as lower survivals keep fish from reaching and staying at large breeding sizes, with higher fecundity. The transition size to breeding female also drives uncertainty in LEP - the higher the transition size to female, the less time the fish has at a size where its reproduction is counted in LEP. 

\begin{figure}[H] % Uncertainty in LEP
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/LEP_uncertainty_breakdown.pdf}}
	\caption{The contribution of different sources of uncertainty in LEP. \label{APP_FIG_Uncertainty_LEP}}
\end{figure}

\paragraph{Lifetime recruit production (LRP)}

Most of the uncertainty in LRP comes from uncertainty in the size of a recruit. This is an artifact of our sampling, where we are unable to estimate egg-recruit survival differently to account for changes in the size of a recruit, so raising the size of a recruit reduces the mortality included in LRP without increasing the mortality included in egg-recruit survival, as it should in an ideal situation.

\begin{figure}[H] % Uncertainty in LEP_R - check plots, weird that assigned offspring and prob_r are just points (unless I don't understand how violins work? Shouldn't they be essentially horizonal lines if all runs have the same value, kind of like growth? Also, why does breeding size have such a different shape for LRP and LEP? Seems weird...)
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/LEP_R_uncertainty_breakdown.pdf}}
	\caption{The contribution of different sources of uncertainty in LRP. \label{APP_FIG_Uncertainty_LEP_R}}
\end{figure}

\begin{figure}[H] % Uncertainty in LEP_R with DD accounted for
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/LEP_R_uncertainty_breakdown_with_DD.pdf}}
	\caption{The contribution of different sources of uncertainty in LRP, when we account for density-dependence in egg-recruit survival. \label{APP_FIG_Uncertainty_LEP_R_DD}}
\end{figure}

\paragraph{Egg-recruit survival ($S_e$)}
\begin{figure}[H] % Uncertainty in recruits-per-egg
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/RperE_uncertainty_breakdown.pdf}}
	\caption{The contribution of different sources of uncertainty in egg-recruit survival. \label{APP_FIG_Uncertainty_RperE}}
\end{figure}

\begin{figure}[H] % Uncertainty in recruits-per-egg with DD accounted for
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/RperE_uncertainty_breakdown_DD.pdf}}
	\caption{The contribution of different sources of uncertainty in egg-recruit survival when we account for density-dependence in egg-recruit survival. \label{APP_FIG_Uncertainty_RperE_DD}}
\end{figure}

\paragraph{Network persistence (NP)}
\begin{figure}[H] % Uncertainty in NP
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/NP_uncertainty_breakdown.pdf}}
	\caption{The contribution of different sources of uncertainty in NP. \label{APP_FIG_Uncertainty_NP}}
\end{figure}

\begin{figure}[H] % Uncertainty in NP with DD accounted for
	\centering
	\includegraphics[width = 1.0\textwidth]{\detokenize{../Plots/FigureDrafts/NP_uncertainty_breakdown_DD.pdf}}
	\caption{The contribution of different sources of uncertainty in NP when we account for density-dependence in egg-recruit survival. \label{APP_FIG_Uncertainty_NP_DD}}
\end{figure}

\newpage{}

%\bibliography{../../../BibTexReferences}
\bibliography{BibTexReferences}
\bibliographystyle{plainnat}

\end{document}